check_binary.sh

```diff
@@ -0,0 +1,194 @@
+      nm "$libtorch" | c++filt | grep cxx11
+    fi
+    exit 1
+  else
+    echo "cxx11 symbols seem to be in order"
+  fi
+fi # if on Darwin
+
+###############################################################################
+# Check for no OpenBLAS
+# TODO Check for no Protobuf symbols (not finished)
+# Print *all* runtime dependencies
+###############################################################################
+# We have to loop through all shared libraries for this
+if [[ "$(uname)" == 'Darwin' ]]; then
+  all_dylibs=($(find "$install_root" -name '*.dylib'))
+  for dylib in "${all_dylibs[@]}"; do
+    echo "All dependencies of $dylib are $(otool -L $dylib) with rpath $(otool -l $dylib | grep LC_RPATH -A2)"
+
+    # Check that OpenBlas is not linked to on Macs
+    echo "Checking the OpenBLAS is not linked to"
+    if [[ -n "$(otool -L $dylib | grep -i openblas)" ]]; then
+      echo "ERROR: Found openblas as a dependency of $dylib"
+      echo "Full dependencies is: $(otool -L $dylib)"
+      exit 1
+    fi
+
+    # Check for protobuf symbols
+    #proto_symbols="$(nm $dylib | grep protobuf)" || true
+    #if [[ -n "$proto_symbols" ]]; then
+    #  echo "ERROR: Detected protobuf symbols in $dylib"
+    #  echo "Symbols are $proto_symbols"
+    #  exit 1
+    #fi
+  done
+else 
+  all_libs=($(find "$install_root" -name '*.so'))
+  for lib in "${all_libs[@]}"; do
+    echo "All dependencies of $lib are $(ldd $lib) with runpath $(objdump -p $lib | grep RUNPATH)"
+
+    # Check for protobuf symbols
+    #proto_symbols=$(nm $lib | grep protobuf) || true
+    #if [[ -n "$proto_symbols" ]]; then
+    #  echo "ERROR: Detected protobuf symbols in $lib"
+    #  echo "Symbols are $proto_symbols"
+    #  exit 1
+    #fi
+  done
+fi
+
+
+###############################################################################
+# Check simple Python calls
+###############################################################################
+if [[ "$PACKAGE_TYPE" == 'libtorch' ]]; then
+  # For libtorch testing is done. All further tests require Python
+  exit 0
+fi
+python -c 'import torch'
+python -c 'from caffe2.python import core'
+
+
+###############################################################################
+# Check for MKL
+###############################################################################
+if [[ "$(uname)" != 'Darwin' || "$package_type" != *wheel ]]; then
+    echo "Checking that MKL is available"
+    python -c 'import torch; exit(0 if torch.backends.mkl.is_available() else 1)'
+fi
+
+
+###############################################################################
+# Check CUDA configured correctly
+###############################################################################
+# Skip these for Windows machines without GPUs
+if [[ "$OSTYPE" == "msys" ]]; then
+    GPUS=$(wmic path win32_VideoController get name)
+    if [[ ! "$GPUS" == *NVIDIA* ]]; then
+        echo "Skip CUDA tests for machines without a Nvidia GPU card"
+        exit 0
```

Comment:
perhaps just set `DESIRED_CUDA='cpu'`?

Reply:
@peterjc123 has context on this. I only copied this from another file